app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{email.downcase!}
+  validates :name,  presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+  format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```

Comment:
1. thứ tự các macro bị sai coding convention, em check lại nhé
https://github.com/framgia/coding-standards/blob/master/vn/rails/standard.md#activerecord

2. Phân biệt `downcase!` và `downcase` giúp chị

3. các giá trị như 50 255 6
thì em khai báo vào file Settings nhé
tìm hiểu về `gem config in rails ` nhé

Reply:
- downcase và downcase! sẽ trả thay đổi xâu ký tự hoa thành thường
- downcase sẽ trả về giá trị bình thường nếu như xâu ký tự không có thay đổi.
     + Ví dụ:  a = "abc"
                   a.downcase # return: abc
- Khi sử dụng downcase thì giá trị sẽ không bị thay đổi
     + Ví dụ:   a = "ABC"
                   a.downcase # return: abc
                   puts a # return: ABC
- downcase! sẽ trả về nil nếu như xâu ký tự không có thay đổi.
     + Ví dụ:  a = "abc"
                   a.downcase # return: nil
- Khi sử dụng downcase! thì giá trị sẽ bị thay đổi
     + Ví dụ:  a = "ABC"
                   a.downcase! # return: abc
                   puts a # return: abc